### DISCUSSION TITLE
Adding custom milter to master.cf

### DISCUSSION THREAD

**Discussion by ginkel** [docker-mailserver/docker-mailserver#4223] [Q&A] [answered]
Hi there,

how would I add a custom milter to `master.cf`? I read about `postfix-master.cf` and that it is not appended literally. How would I append the following (still untested) snippet?

```
opengpg-filter   unix  -       n       n       -       10      pipe
  flags=Rq user=postfix argv=/usr/bin/python3 /tmp/docker-mailserver/opengpg-milter.py
```

Thanks,
Thilo

**Comment by polarathene** [comment, marked as answer]
https://github.com/docker-mailserver/docker-mailserver/issues/4218#issuecomment-2400701472

```bash
postconf -M 'opengpg-filter unix - n n - 10 pipe flags=Rq user=postfix argv=/usr/bin/python3 /tmp/docker-mailserver/opengpg-milter.py'
```

See the workaround section at the end of [this comment](https://github.com/docker-mailserver/docker-mailserver/issues/4220#issuecomment-2403272740) for reference of adding a `user-patches.sh` to your config volume. That will run the command when starting a new DMS container instance.

It should add the line to `/etc/postfix/master.cf` for you. `postfix-master.cf` is only used to override settings.
